server/base/src/test/java/org/apache/accumulo/server/ServerContextTest.java

```diff
@@ -135,7 +135,9 @@ public void testCanRun() {
     // ensure this fails with older versions; the oldest supported version is hard-coded here
     // to ensure we don't unintentionally break upgrade support; changing this should be a conscious
     // decision and this check will ensure we don't overlook it
-    final int oldestSupported = AccumuloDataVersion.ROOT_TABLET_META_CHANGES;
+
+    // TODO basically disable check until upgrade to 3.1 is supported.
+    final int oldestSupported = AccumuloDataVersion.METADATA_FILE_JSON_ENCODING;
```

Comment:
Instead of doing this, you should use `@Disabled`

```suggestion
    final int oldestSupported = AccumuloDataVersion.ROOT_TABLET_META_CHANGES;
```

On the method, add:


```java
@Disabled("must fix https://github.com/apache/accumulo/issues/3768 to re-enable")
```

Reply:
I thought of using disabled - opted to make this test fail instead.  This way, as the code is changed to support the necessary version, this test will fail and be impossible to overlook.

Reply:
When you say you opted to make this test fail instead, do you mean when the fix for #3768 is eventually upgraded? It looks like you have it set to pass right now, but should fail after the upgrade is done on line 146 with:


```java
    IntStream.of(oldestSupported - 1, currentVersion + 1).forEach(shouldFail);
```

Is that what you mean?
If so, I think that's fine. Otherwise, I'm confused.

Reply:
Yes.  When adding the list of supported versions in `UpgradeCoordinator` the forEach(should fail) kicks out (line my off by one with my version)

```
org.opentest4j.AssertionFailedError: Expected java.lang.IllegalStateException to be thrown, but nothing was thrown.

	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:152)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:73)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:35)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3083)
	at org.apache.accumulo.server.ServerContextTest.lambda$testCanRun$3(ServerContextTest.java:144)
	at java.base/java.util.Spliterators$IntArraySpliterator.forEachRemaining(Spliterators.java:1032)
	at java.base/java.util.stream.IntPipeline$Head.forEach(IntPipeline.java:593)
	at org.apache.accumulo.server.ServerContextTest.testCanRun(ServerContextTest.java:147)
	...
```

Reply:
Yes - when adding the supporting versions to `UpgradeCoordinator` the line 
```
 IntStream.of(oldestSupported - 1, currentVersion + 1).forEach(shouldFail);
```
causes a hard fail.  (My line numbers may be off by one.)

```
org.opentest4j.AssertionFailedError: Expected java.lang.IllegalStateException to be thrown, but nothing was thrown.

	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:152)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:73)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:35)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3083)
	at org.apache.accumulo.server.ServerContextTest.lambda$testCanRun$3(ServerContextTest.java:144)
	at java.base/java.util.Spliterators$IntArraySpliterator.forEachRemaining(Spliterators.java:1032)
	at java.base/java.util.stream.IntPipeline$Head.forEach(IntPipeline.java:593)
	at org.apache.accumulo.server.ServerContextTest.testCanRun(ServerContextTest.java:147)
```

Reply:
Okay, in that case, I think it would be more clear that it is disabled by referencing the current version instead of the name of the version. Also, can add the commented out version that it needs to go back to when it is fixed later.

```suggestion
    // TODO basically disable check until upgrade to 3.1 is supported.
    // final int oldestSupported = AccumuloDataVersion.ROOT_TABLET_META_CHANGES;
    final int oldestSupported = AccumuloDataVersion.CURRENT_VERSION;
```

Reply:
I don't think you included my suggestion to make the test point directly to the `CURRENT_VERSION` alias. You left it pointing to `METADATA_FILE_JSON_ENCODING`. I know they are the same value, but I think that leaving it pointing to the specific named version implies there's some reason that specific version was picked. But the reason it was picked is because it's the current version. So, it's more clear to just directly point to `CURRENT_VERSION`. But it doesn't matter that much, because this should all get backed out once the upgrade code is added, so I don't think it's worth changing now. I just wanted to explain my reasoning a bit more.